server/app/lib/actions/fusor/deployment/cloud_forms/update_root_password.rb

```diff
@@ -26,47 +26,69 @@ def plan(deployment, vm_ip)
           end
 
           def run
-            Rails.logger.info "================ UpdateRootPassword run method ===================="
+            begin
+              Rails.logger.info "================ UpdateRootPassword run method ===================="
 
-            ssh_user = "root"
-            ssh_password = "smartvm"
+              ssh_user = "root"
+              ssh_password = "smartvm"
 
-            deployment = ::Fusor::Deployment.find(input[:deployment_id])
+              deployment = ::Fusor::Deployment.find(input[:deployment_id])
 
-            # TODO: observing issues with running the appliance console using SSHConnection; therefore, temporarily
-            # commenting out and using the approach above which will run it from a python script
-            @success = false
-            @io = StringIO.new
-            client = Utils::Fusor::SSHConnection.new(input[:vm_ip], ssh_user, ssh_password)
-            client.on_complete(lambda { update_root_password_completed })
-            client.on_failure(lambda { update_root_password_failed })
-            cmd = "echo \"#{deployment.cfme_root_password}\" | passwd --stdin #{ssh_user}"
-            client.execute(cmd, @io)
+              # TODO: observing issues with running the appliance console using SSHConnection; therefore, temporarily
+              # commenting out and using the approach above which will run it from a python script
+              @success = false
+              @retry = false
+              @io = StringIO.new
+              client = Utils::Fusor::SSHConnection.new(input[:vm_ip], ssh_user, ssh_password)
+              client.on_complete(lambda { update_root_password_completed })
+              client.on_failure(lambda { update_root_password_failed })
+              cmd = "echo \"#{deployment.cfme_root_password}\" | passwd --stdin #{ssh_user}"
+              client.execute(cmd, @io)
 
-            Rails.logger.info "================ Leaving UpdateRootPassword run method ===================="
+              Rails.logger.info "================ Leaving UpdateRootPassword run method ===================="
+              # close the stringio at the end
+              @io.close unless @io.closed?
+              if not @success and @retry
```

Comment:
Although it will not make a difference on this line, I'd generally recommend using && versus 'and'.  The 'and' operator has lower precedence than && which often results in unexpected behavior.